[#2] 직원 검색 및 조회 구현

chaewon02 opened the pull request:
- Ability many to one 수정
- 직원 이름, 사번 : keyword / 직원 소속 부서 : department / 직원 개인 역량 : ability 에 따른 직원 검색
- 특정 직원이 갖는 ability list 조회

chaewon02 committed:
feat: 직원 검색 및 조회 구현

- Ability many to one 수정
- 직원 이름, 사번 : keyword / 직원 소속 부서 : department / 직원 개인 역량 : ability 에 따른 직원 검색
- 특정 직원이 갖는 ability list 조회

OrangeVinyl commented on src/main/java/org/finalpjt/hraccoon/domain/user/ctrl/UserController.java:
"`**/user/search**`" method 에서 RequestParam을 여러 개 말고 하나로 통합하는 방법은 없나요?

OrangeVinyl commented on src/main/java/org/finalpjt/hraccoon/domain/user/repository/AbilityRepository.java:
```diff
@@ -0,0 +1,21 @@
+	@Query("select a from Ability a where a.user.userId = :userId")
+	List<Ability> findByUserId(String userId);
+
+	//안 쓸듯
```
안 쓸 것 같은 추상 메서드는 지워주세요 :)

OrangeVinyl commented on src/main/java/org/finalpjt/hraccoon/domain/user/service/UserService.java:
사용하지 않는 주석은 지워주세요.

OrangeVinyl commented on src/main/java/org/finalpjt/hraccoon/domain/user/data/dto/response/UserSearchResponse.java:
```diff
@@ -0,0 +1,43 @@
+	private String userEmail;
+
+	/* UserDetail */
+	private Integer userRemainVacation;
```
위 dto에서 해당 변수를 사용하는 일이 없는 것 같은데 선언하신 이유가 있나요?

chaewon02 committed:
refactor: 피드백 적용

- UserSearchResponse.java 안 쓰는 변수 삭제
- UserService.java 주석 삭제
- AbilityRepository.java 주석 삭제 (사용중인 메서드)

chaewon02 committed:
feat: 팀별 일정 조회 (팀 캘린더)

- ApprovalRepository.java, ApprovalResponse.java 추가 및 수정
- UserController.java 에서 getTeamApprovalInfo 메서드 추가
